src/applications/bmqtool/m_bmqtool_application.cpp

```diff
@@ -1046,12 +1052,7 @@ void Application::producerThread()
         }
     }
 
-    // Finished posting messages in auto mode?
-    // If shutDownGrace is set, signal to the main thread to exit.
-    if (d_parameters_p->mode() == ParametersMode::e_AUTO &&
-        d_parameters_p->shutdownGrace() != 0) {
-        // We do not need to sleep the grace period, since it is done
-        // by the main thread, in the stop() function.
+    if (!bmqt::QueueFlagsUtil::isAck(d_parameters_p->queueFlags())) {
```

Comment:
Is this going to affect auto consumer?

Reply:
I don't think we can do both at the same time.  So, this does not seem to affect consumer.